### PR TITLE
Promote DaemonRestart e2es out of Flaky

### LINKED ISSUE #17829: e2e flake: DaemonRestart Controller Manager is flaky
The test is in flaky suite and no one seems to care. We need an owner for this test.

cc @wojtek-t @davidopp @ixdy @mikedanese @quinton-hoole 

Last failure log:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/daemon_restart.go:278 Nov 26 03:25:57.460: RcManager created/deleted pods after restart [2015-11-26 03:25:16.775576085 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-yuzjn Phase Running Host e2e-flaky-minion-gtsf [2015-11-26 03:25:38.326844194 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-jwxpg Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:38.379404066 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-9ts1x Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:41.93934037 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-ynst4 Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:41.951709324 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-ynst4 Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:38.321323343 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-9ts1x Phase Pending Host [2015-11-26 03:25:38.42322019 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-575re Phase Pending Host [2015-11-26 03:25:40.869486693 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-575re Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:42.523489077 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-575re Phase Running Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.319163368 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-jm7zt Phase Pending Host [2015-11-26 03:25:38.354714318 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-jwxpg Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:38.462898056 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-ynst4 Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:41.928146952 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-kgn3f Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:42.066522845 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-cs8hq Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:42.078296897 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-cs8hq Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:45.31246032 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-chu1j Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.309458862 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-m4vh5 Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:38.383460818 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-zh1r7 Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:16.775602748 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-yvo1o Phase Running Host e2e-flaky-minion-zprd [2015-11-26 03:25:38.437579768 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-575re Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.475785078 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-575re Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.724975474 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-dxzyf Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:42.860209179 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-m4vh5 Phase Running Host e2e-flaky-minion-zprd [2015-11-26 03:25:38.451727263 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-ynst4 Phase Pending Host [2015-11-26 03:25:38.508389581 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-dxzyf Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:45.318291471 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-chu1j Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.358318412 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-zh1r7 Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:42.084430361 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-cs8hq Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:45.115230633 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-chu1j Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:42.040842918 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-zh1r7 Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:43.124743053 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-9ts1x Phase Running Host e2e-flaky-minion-xrmh [2015-11-26 03:25:16.775473034 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-chu1j Phase Running Host e2e-flaky-minion-xrmh [2015-11-26 03:25:16.775495546 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-diedr Phase Running Host e2e-flaky-minion-gtsf [2015-11-26 03:25:38.421746375 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-kgn3f Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:38.492462125 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-ynst4 Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:38.502379621 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-dxzyf Phase Pending Host [2015-11-26 03:25:41.912051713 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-kgn3f Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:42.05492754 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-zh1r7 Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:38.304511407 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-jwxpg Phase Pending Host [2015-11-26 03:25:44.31217349 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-dxzyf Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:16.775443563 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-1cg0h Phase Running Host e2e-flaky-minion-zprd [2015-11-26 03:25:16.775458239 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-2d2yv Phase Running Host e2e-flaky-minion-xrmh [2015-11-26 03:25:16.775531086 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-vdu88 Phase Running Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.337678073 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-cs8hq Phase Pending Host [2015-11-26 03:25:38.421721822 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-9ts1x Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:41.019190004 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-9ts1x Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:42.048059537 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-zh1r7 Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:38.341308949 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-m4vh5 Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:41.951725731 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-ynst4 Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:44.519906884 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-dxzyf Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.319197645 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-zh1r7 Phase Pending Host [2015-11-26 03:25:16.775427596 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-118w8 Phase Running Host e2e-flaky-minion-gtsf [2015-11-26 03:25:16.775514759 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-uf5ix Phase Running Host e2e-flaky-minion-zprd [2015-11-26 03:25:38.319422668 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-kgn3f Phase Pending Host [2015-11-26 03:25:38.365060848 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-kgn3f Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:41.923586597 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-kgn3f Phase Pending Host e2e-flaky-minion-gtsf [2015-11-26 03:25:43.725212376 -0800 PST] DEL: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-jm7zt Phase Running Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.299112367 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-m4vh5 Phase Pending Host [2015-11-26 03:25:38.344468969 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-jm7zt Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.434080106 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-cs8hq Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:16.775400914 -0800 PST] ADD: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-0zj63 Phase Running Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.369821246 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-jm7zt Phase Pending Host e2e-flaky-minion-xrmh [2015-11-26 03:25:38.393348625 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-cs8hq Phase Pending Host e2e-flaky-minion-zprd [2015-11-26 03:25:44.513737053 -0800 PST] UPDATE: daemonrestart10-24d0f510-9430-11e5-9e6c-42010af01555-dxzyf Phase Pending Host e2e-flaky-minion-xrmh 
```

### PR DESCRIPTION
**Pull request by ikehz**
Fixes #17829 for real.

cc @bprashanth

### CODE REVIEW

File: test/e2e/daemon_restart.go [hunk 1]
```diff
@@ -183,8 +183,7 @@ func getContainerRestarts(c *client.Client, ns string, labelSelector labels.Sele
 	return failedContainers, containerRestartNodes.List()
 }
 
-// Flaky issues #17829, #19023
-var _ = Describe("DaemonRestart [Disruptive] [Flaky]", func() {
+var _ = Describe("DaemonRestart [Disruptive]", func() {
```
Comment (bprashanth) [inline test/e2e/daemon_restart.go]: since all disruptive tests must be run in serial, what's the difference?

Comment (ihmccreery) [inline test/e2e/daemon_restart.go]: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/e2e-tests.md#kinds-of-tests

Comment (bprashanth) [inline test/e2e/daemon_restart.go]: ah, we should totally be tolerant to reastarts in soak-serial but yeah, makes sense

**Comment by davidopp** [comment]
LGTM

Thanks!

File: test/e2e/daemon_restart.go (hunk 1 above)
Comment (ihmccreery) [inline test/e2e/daemon_restart.go]: We should be tolerant, but it defeats the purpose of soak; if we're trying to see what happens when clusters are alive for a long time, bonking them often doesn't help us get there.

Comment (bprashanth) [inline test/e2e/daemon_restart.go]: Ah, you mean don't restart things. I'd assumed disruptive would include all sorts of netsplit, storage loss etc but more likely that it more or less translates to reboot and daemon restart today.

Comment (ihmccreery) [inline test/e2e/daemon_restart.go]: Yes, you're right.  Filed #20491.

**Pull request merged**